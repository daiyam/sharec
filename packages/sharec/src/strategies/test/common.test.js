const path = require('path')
const { readFileSync } = require.requireActual('fs')
const { strategy, yamlStrategy } = require('strategies/common')

describe('strategy > common', () => {
  describe('json strategy', () => {
    const common01 = require('fixtures/common/json/common_01.json')
    const common02 = require('fixtures/common/json/common_02.json')

    it('should merge objects by one deep level', () => {
      expect(strategy(common01, common02)).toMatchSnapshot()
    })
  })

  describe('yaml strategy', () => {
    const common01 = readFileSync(
      path.resolve(
        __dirname,
        '../../../test/fixtures/common/yaml/common_01.yml',
      ),
      'utf8',
    )
    const common02 = readFileSync(
      path.resolve(
        __dirname,
        '../../../test/fixtures/common/yaml/common_02.yml',
      ),
      'utf8',
    )

    it('should merge objects by one deep level', () => {
      expect(yamlStrategy(common01, common02)).toMatchSnapshot()
    })
  })
})
