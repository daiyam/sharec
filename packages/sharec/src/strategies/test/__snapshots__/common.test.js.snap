// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`strategy > common json strategy should merge objects by one deep level 1`] = `
Object {
  "bar": Object {
    "foo": "foo",
  },
  "baz": Object {
    "bar": "baz",
    "foo": "bar",
  },
  "foo": "baz",
}
`;

exports[`strategy > common yaml strategy should merge objects by one deep level 1`] = `
"foo: baz
bar:
  - foo: foo
baz:
  - foo: bar
    bar: baz
"
`;
